### PR TITLE
MQTTv5 Packet bug fixes and samples

### PR DESCRIPTION
**Pull request by CIPop**
- Adding new MQTTV5Packet samples (derived from existing v3 samples). The new samples showcase topic aliasing, user properties and *ACK properties.
- BugFix: User Properties bug fix (picked commit from #234, maintained commit user assignment /cc @MrEthan). Includes Unit Test changes.
- BugFix: User Properties bug fix to exclude NULL characters [MQTT-1.5.4-2] from the [MQTTv5 OASIS specification](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901010). Includes Unit Test changes.
- BugFix: MQTTv5 reason code should not be extended as a negative int. Includes Unit Test changes.
- Samples enhancement: string representation of MQTTv5 reason codes and properties.
- Other small changes:
  - Excluding vscode, doxygen generated files from gitignore.
  - CMake build of the Linux MQTT Packet Samples.
  - Switched all samples to use the publicly available [test.mosquitto.org](https://test.mosquitto.org) MQTTv3/v5 broker.
  - Added instructions for running the tests locally.

### CODE REVIEW

File: MQTTPacket/test/test2.c [hunk 1]
```diff
@@ -890,7 +893,8 @@ int test9(struct Options options)
 	int rc = 0;
 	unsigned char buf[100];
 	int buflen = sizeof(buf);
-	int reasonCode = 33, reasonCode2 = 44;
+	int reasonCode = BAD_AUTHENTICATION_METHOD; // // (140) test signed char to unsigned char conversion.
```
Comment (vaavva) [inline MQTTPacket/test/test2.c:896]:
```suggestion
	int reasonCode = BAD_AUTHENTICATION_METHOD; // (140) test signed char to unsigned char conversion.
```

**Review by vaavva — APPROVED** [review]

File: MQTTPacket/test/test2.c (hunk 1 above)
Comment (CIPop) [reply MQTTPacket/test/test2.c:896]: Thanks!

**Comment by CIPop** [comment]
@icraggs I have verified that the original PR#234 _does_ have the ECA requirement. 
Merging the PRs in order should automatically resolve any conflicts. If not, I'll be rebasing mine once #234 is merged.

**Review by icraggs — CHANGES_REQUESTED** [review]
HI Cristian. The changes look good. But the copyright headers need fixing up. In the new files, the copyright statement should be:

Copyright (c) 2023 (your name or company name)

and in the updated files:

Copyright (c) 2014, 2023 IBM Corp. 

(the start year should be unchanged). If you add significant content you might like to add your name or company name at the end too:

Copyright (c) 2014, 2023 IBM Corp., Cristian Pop

Thanks

**Comment by CIPop** [comment]
Thanks for the guidance @icraggs! I will make the modifications by EOD today.

Since it is acceptable, I will be removing my name from all file headers: I was complying with project patterns, but my company prefers scrubbing developer names from source code.

**Comment by icraggs** [comment]
> Thanks for the guidance @icraggs! I will make the modifications by EOD today.
> 
> Since it is acceptable, I will be removing my name from all file headers: I was complying with project patterns, but my company prefers scrubbing developer names from source code.

When I was working for IBM, IBM held the copyright, which is why their name is on the headers. Now I work for myself, so I can use my own name :-) It's whatever your company prefers.

**Comment by CIPop** [comment]
@icraggs I've updated the copyright notices - PTAL to ensure this is indeed the correct form.

**Comment by CIPop** [comment]
Thank you @icraggs ! I will split up PRs in the future. We were aware and did expect the additional review (this has been discussed in the last IoT WG meeting as well).

I will be out on vacation for 2 weeks: I will continue to monitor the issue if changes are required, but there will be some delays.

**Pull request merged**